### DISCUSSION TITLE
The purpose of `io.Copy(ioutil.Discard, r)` in `render.DecodeJSON`

### DISCUSSION THREAD

**Discussion by haarts** [go-chi/chi#871] [Q&A] [answered]
In the function [`DecodeJSON`](https://github.com/go-chi/render/blob/v1.0.3/decoder.go#L42-L45) this is called:

```go
io.Copy(ioutil.Discard, r)
```

what is the purpose of that?

**Comment by VojtechVitek** [comment, marked as answer]
This is the reason: https://github.com/go-chi/render/issues/42

`json.NewDecoder(r).Decode(v)` reads the first JSON object only and leaves the rest of the bytes unread in the reader.

**Reply by haarts** [reply]
Fascinating. Thank you.
